### PR TITLE
Support for Multiple StorageManager within Spring Boot 2.x integration. Fixes #489

### LINKED ISSUE #489: Support multiple StorageManagers within Spring Boot Integration
It should be possible to support multiple StorageManagers within the integration using Qualifiers.

Due to limitations in looking up annotations on injection point from within Bean producer method, some developer actions are required. 

```
@Configuration
public class DefineStorageManagers {

    private final StorageManagerProvider provider;

    public DefineStorageManagers(StorageManagerProvider provider) {
        this.provider = provider;
    }

    @Bean
    @Qualifier("green")
    public EmbeddedStorageManager getGreenManager() {
        return provider.get(DatabaseColor.GREEN.getName());
    }
    @Bean
    @Qualifier("red")
    public EmbeddedStorageManager getRedManager() {
        return provider.get(DatabaseColor.RED.getName());
    }
}
```

The qualifier (label) is used within the configuration key

```
one.microstream.red.storage-directory=red-db
one.microstream.red.channel-count=2

one.microstream.green.storage-directory=green-db
one.microstream.green.channel-count=1
```

The automatic Root bean creation (and injection), `EmbeddedStorageFoundationCustomizer`, and `StorageManagerInitializer` are supported.

### PR DESCRIPTION
**Pull request by rdebusscher**
Draft: Need some more testing and documentation

### CODE REVIEW

**Comment by rdebusscher** [comment]
See extended example at https://github.com/rdebusscher/microstream-spring-boot-patterns/tree/main/v2/multiple-managers

File: integrations/spring-boot/src/it/primary/src/main/java/test/microstream/integrations/spring/boot/service/NamesService.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ * #L%
+ */
+
+import org.springframework.stereotype.Controller;
+import test.microstream.integrations.spring.boot.database.Names;
+
+import java.util.Collection;
+
+@Controller
```
Comment (zdenek-jonas) [inline integrations/spring-boot/src/it/primary/src/main/java/test/microstream/integrations/spring/boot/service/NamesService.java:28]: It is controler or service?
@Controller oder @Service?

File: integrations/spring-boot/src/it/multi-manager/src/main/java/test/microstream/integrations/spring/boot/RootPreparationOfGreenDatabase.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import test.microstream.integrations.spring.boot.database.Names;
+
+@Component
+public class RootPreparationOfGreenDatabase implements StorageManagerInitializer {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RootPreparationOfGreenDatabase.class);
+
+    @Override
+    public void initialize(final StorageManager storageManager) {
+        if (!DatabaseColor.GREEN.getName().equals(storageManager.databaseName())) {
```
Comment (zdenek-jonas) [inline integrations/spring-boot/src/it/multi-manager/src/main/java/test/microstream/integrations/spring/boot/RootPreparationOfGreenDatabase.java:38]: Isn't there any way to solve this through the qualifier?

File: integrations/spring-boot/src/it/primary/src/main/java/test/microstream/integrations/spring/boot/service/NamesService.java (hunk 1 above)
Comment (rdebusscher) [reply integrations/spring-boot/src/it/primary/src/main/java/test/microstream/integrations/spring/boot/service/NamesService.java:28]: Service is probably better, I'll update.

File: integrations/spring-boot/src/it/multi-manager/src/main/java/test/microstream/integrations/spring/boot/RootPreparationOfGreenDatabase.java (hunk 2 above)
Comment (rdebusscher) [reply integrations/spring-boot/src/it/multi-manager/src/main/java/test/microstream/integrations/spring/boot/RootPreparationOfGreenDatabase.java:38]: I get a list of all beans implementing StorageManagerInitializer from Spring. I can have a look at the Class definition and look to see if there is a Qualifier or not and it that matches to know if I need to call the initialize method for that StorageManager.

I'll have a look to implement that solution also.

**Review by zdenek-jonas — APPROVED** [review]

**Pull request merged**